mautic/mautic
# Removing calendar bundle

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [n]
| New feature/enhancement? (use the a.x branch)      | [y]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [y]
| Automated tests included?              | [n] 
| Related user documentation PR URL      | TODO
| Related developer documentation PR URL | TODO
| Issue(s) addressed                     | Fixes https://forum.mautic.org/t/is-anyone-using-mautics-built-in-calendar/23638/20



#### Description:

As the Calendar in Mautic is [useless](https://github.com/mautic/mautic/issues/9964) and [insecure](https://github.com/mautic/mautic/issues/9022) we [decided to remove it](https://forum.mautic.org/t/is-anyone-using-mautics-built-in-calendar/23638) to save on the maintenance work, speed up the application and reduce the asset size. 

The Calendar takes valuable space in the main Mautic menu as the second item:
![Screen Shot 2022-07-18 at 15 46 45](https://user-images.githubusercontent.com/1235442/179525549-60256728-7a18-4d94-bc93-e7053881d531.png)

Yet it bring little to no value to the users. If anyone tries to use it then they find out the information displayed in it is incomplete and therefore useless. 

Yes, it can be improved so it would be useful and for example help users to use Mautic as CRM but that is not the goal of Mautic.

When you [look at the commit history](https://github.com/mautic/mautic/commits/5.x/app/bundles/CalendarBundle) of the Calendar Bundle it had been touched since 2016 only for maintenance and bug fixes work. So it costs valuable time of Mautic developers for no value added to the project.

The asset size of CSS and JS files will be reduced by nearly 8%:
```
media/css/app.css: 135 KB -> 134 KB
media/js/app.js: 414 KB -> 392 KB
media/js/libraries.js: 2041 KB -> 1863 KB
```

And it will increase our test coverage by 2.3% 🚀 

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Make sure the Calendar is gone and everything else is working as before.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11270"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
